### PR TITLE
Fix Firestore sometimes not using emulator and app leaks.

### PR DESCRIPTION
**Pull request by yuchenshi**
## Problem

Sometimes when Emulator Suite is restarted while on the Firestore tab, `useEmulator()` is sometimes not correctly called on the Firestore instance. It's not easy to reproduce, but can be done by killing `firebase emulators:start` with `Ctrl-C` and run the command again, and again. (In my experience, it depends on the polling timing and usually happens around the second or third time).

## The fix

I've tried to fix this in various ways, but the only way I can be sure of is to move the `useEmulator` call right next to the app initialization. That's also the recommended timing anyway and we don't need to reason about async ordering. This also fixes the leaking of app instances (i.e. more and more apps will keep being created and never deleted without the fix).

Firestore SDK lazy loading has been disabled for now. Sorry, but that just makes debugging why such things happen much harder and may be part of the cause. Feel free to investigate and add it back in the future with adequate testing.

### CODE REVIEW

File: src/firebase.ts [hunk 1]
```diff
@@ -65,23 +66,33 @@ export const getStorageApp = (host: string, port: string) => {
...
 
-export function useEmulatedFirebaseApp(name: string, config: any) {
+export function useEmulatedFirebaseApp(
+  name: string,
+  config: any,
+  initialize?: (app: firebase.app.App) => void
+) {
   const projectId = useProjectId();
-
-  const app = useMemo(() => {
```
Comment (yuchenshi) [inline src/firebase.ts:71]: Hooks are called during render phase and should not have side effects like this. Wrapping side effects in `useMemo` won't help. The right way is to `useEffect` for side effects and pair them up with cleanups.

To see the total active apps, run `firestore.app.firebase_.apps` in console. You'll see twice as many apps being created than deleted without the fix.

File: src/components/Firestore/index.tsx [hunk 2]
```diff
@@ -88,11 +81,6 @@ export const Firestore: React.FC = React.memo(() => {
   const showCollectionShell = path.split('/').length < 2;
   const showDocumentShell = path.split('/').length < 3;
 
-  useEffect(() => {
```
Comment (yuchenshi) [inline src/components/Firestore/index.tsx:91]: This is a duplicate of `FirestoreDevTools`.

**Comment by yuchenshi** [comment]
DO NOT MERGE: Just noticed that this causes the app to be recreated on every router navigation. Investigating.

File: src/firebase.ts [hunk 3]
```diff
@@ -65,23 +66,49 @@ export const getStorageApp = (host: string, port: string) => {
   return app;
 };
 
-export function useEmulatedFirebaseApp(name: string, config: any) {
+/**
+ * Get an JS SDK App instance with emulator Admin auth enabled.
```
Comment (sam-gc) [inline src/firebase.ts:70]: nit "a" not "an"

**Review by sam-gc — APPROVED** [review]

**Review by NothingEverHappens — APPROVED** [review]

File: src/firebase.ts (hunk 3 above)
Comment (yuchenshi) [reply src/firebase.ts:70]:
```suggestion
 * Get a JS SDK App instance with emulator Admin auth enabled.
```

**Pull request merged**